consumer/src/main/java/com/flipkart/varadhi/consumer/ConsumerStatus.java

```diff
@@ -0,0 +1,18 @@
+package com.flipkart.varadhi.consumer;
+
+/**
+ * Status for a consumer that has been started.
+ */
+public enum ConsumerStatus {
+
+    CONSUMING,
+
+    ERRORED,
+
+    /**
+     * Throttled is a wide bucket. TODO: evaluate splitting based on the requirements later on.
+     */
+    THROTTLED,
+
+    PAUSED
```

Comment:
I actually think "pause" meaning stop message delivery immediately as a nice feature. It has nothing to do with allocation. 
Basically an api, that you can use to "pause" message delivery and which works immediately.

we can also think of downgrading extended "pause" duration to "stopped" state.

Reply:
`subscription is running, but not consuming as there is nothing to consume` - it will be CONSUMING state.
`CONSUMING` is a "state" in which the consumer is in. When in "Consuming", the varadhi consumer will be attempting message delivery for un-delivered messages.
But if there are no undelivered messages, then obviously the destination wont get any. The consume rate can be fetched orthogonally from the stats exposed by the varadhi consumer.  

PAUSED - means that consumer is intentionally not delivering any messages.